### DISCUSSION TITLE
Can I download backup locally?

### DISCUSSION THREAD

**Discussion by mbrain-io** [planetscale/discussion#168] [Q&A] [answered]
Hello guys, awesome product! Trying it for the first time with Prisma and I am really impressed :-)

Btw how can I download backup and save it locally?

**Comment by fooforge** [comment, marked as answer]
Hey @mbrain-io,

Our [`pscale` CLI tool](https://github.com/planetscale/cli) allows you to dump a database like so:

```
pscale database dump [database] [branch]
```

Executing the above command will create a folder in your current working directory with (at least) two SQL dumps per table: One for the table's schema and one for the actual data stored in the table.

Another way to achieve the same would be to open a secure connection to your database. Like so:

```
❯ pscale connect [database] [branch]
Tried address 127.0.0.1:3306, but it's already in use. Picking up a random port ...
Secure connection to database [database] and branch [branch] is established!.

Local address to connect your application: 127.0.0.1:46654 (press ctrl-c to quit)
```

With the socket address you can then connect with a MySQL client in another terminal session, e.g.

```
❯ mysql -u root -h 127.0.0.1 -P 46654
```

This could, of course, also be a mysqldump or any other data manipulation tool you would like to connect with:

```
❯ mysqldump -u root -h 127.0.0.1 -P 46554 [database] > planetscale-backup-[database]-[branch]-`date +%s`.sql
```

Cheers,
Mike

**Reply by mi5ha** [reply]
@fooforge Awesome tnx!

**Reply by miyapt** [reply]
mysql> mysqldump -u root -h 127.0.0.1 -P 5116 [dtabase]
    -> ;
ERROR 1105 (HY000): syntax error at position 10 near 'mysqldump'

**Reply by fooforge** [reply]
@miyapt Looks like you have also created a ticket with us. I've responded to it a few minutes ago. In short: `mysqldump` is its own binary. You'll need to run it from a terminal, not from within a `mysql` session.

**Reply by ireneeromero** [reply]
I have used `pscale database dump [database] [branch]` but it only downloaded the schema of each table.  Where is the data or how can I fix it?
 
![image](https://github.com/planetscale/discussion/assets/61683312/070c9c7b-8624-4032-880a-1df951c0695a)

**Reply by jgreet** [reply]
Is your data in the **main** branch? Looking at the database named enbic2-lab it appears most of the activity is in a branch named **initial-setup**. Try `pscale db dump enbic2-lab initial-setup`.

**Reply by ireneeromero** [reply]
You are right! It was my confusion, all tghe data was in initial-setup branch. 

Thank you!
